src/util/route.js

```diff
@@ -6,13 +6,13 @@ export function isSameRoute (a: Route, b: ?Route): boolean {
   } else if (a.path && b.path) {
     return (
       a.path === b.path &&
-      a.hash === b.hash &&
+      (a.hash || '') === (b.hash || '') &&
```

Comment:
Location can be an empty object because everything is optional in it.
Is there a way on flow to say a Location must have either a name or a path?
I think using multiple types and then saying location is one of them can work